test/simplification/required.spec.ts

```diff
@@ -0,0 +1,100 @@
+import simplifyRequired from '../../src/simplification/SimplifyRequired';
+
+/**
+ * Some of these test are purely theoretical and have little if any merit 
+ * on a JSON Schema which actually makes sense but are used to test the principles.
+ */
+describe('Simplification of required', function() {
+  test('should support simple required', function() {
+    const schema = {
+      "type": "object",
+      "properties": { 
+        "testProp1": { 
+          "type": "string"
+        } 
+      },
+      "required": ["testProp1"],
+    }
+    const expected = ["testProp1"];
+
+    const required = simplifyRequired(schema);
+    expect(required).toEqual(expected);
+  });
+
+  test('should support advanced required with conditions', function() {
+    const schema = {
+      "type": "object",
+      "properties": { 
+        "testProp1": { 
+          "type": "string"
+        } 
+      },
+      "required": ["testProp1"],
+      "if": {
+          "properties": { 
+            "testProp1": { 
+              "const": "string"
+            } 
+          } 
+      },
+      "then": {
+          "properties": { 
+            "testProp2": { 
+              "type": "string"
+            } 
+          } 
+      },
+      "else": {
+          "properties": { 
+            "testProp3": { 
+              "type": "string"
+            } 
+          },
+          "required": ["testProp3"],
+      }
+    }
```

Comment:
Just a suggestion, but I find it more easy to read both test wise and input if the larger schemas reside in an external file.

Reply:
but simplification of required is not complex like other simplification :) and also I don't want use mocks, but check output value.